org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/WorkspaceStateWriter.java

```diff
@@ -46,7 +46,7 @@
  * Maintains map file of maven artifacts present in workspace.
  */
 
-@Component(service = {IMavenProjectChangedListener.class})
+@Component(service = {IMavenProjectChangedListener.class}, immediate = true)
```

Comment:
I think we better should have a component consuming hte service than making it immedate..

Reply:
Then which one ? The bundle ? Better defining is as an extension that through DS then